Improvements, adjustments

MikeGamePro opened the pull request:
- Renamed 'dax' to 'nodax'
- Added commands 'cam' and 'press'
- Gave 'rjto' and 'iframes' durations
- Changed the no-look-around time from 3 seconds to 5 for 'stickycam'
- Added checks for 'cd', 'dur', 'enable', and 'disable' to see if the command exists
- 'bighead', 'smallhead', and 'hugehead' now disable each other
- Added mod checks for the camera movement commands (turn-left, turn-180, etc)
- Started something for closing goalc when gk is closed but can't figure out how to properly close goalc, so will come back to this (line 789)
- env default value adjustments

MikeGamePro committed:
Improvements, adjustments

- Renamed 'dax' to 'nodax'
- Added 'cam' and 'press'
- Gave 'rjto' and 'iframes' durations
- Changed the no-look-around time from 3 seconds to 5 for 'stickycam'
- Added checks for 'cd', 'dur', 'enable', and 'disable' to see if the commands exist
- 'bighead', 'smallhead', and 'hugehead' now disable each other
- Added mod checks for the camera movement commands (turn-left, turn-180, etc)
- Started something for closing goalc when gk is closed but can't figure out how to properly close goalc, so will come back to this (line 789)
- env default value adjustments

dallmeyer commented on resources/twitchcommands.py:
```diff
@@ -443,8 +445,13 @@ def gamecontrol():
 
+        if PREFIX + "cam" == str(args[0]).lower() and len(args) >= 2 and on_check("cam") and cd_check("cam"):
+            activate("cam")
+            sendForm("(send-event *camera* 'change-state cam-" + str(args[1]) + " 0)(send-event *target* 'no-look-around (seconds " + str(durations[command_names.index("cam")]) + "))")
```
hopefully this just does nothing if the user gives a bad cam mode?

MikeGamePro replied:
yeah it just gives a repl error and nothing breaks from what I've seen. It still starts the cooldown but that's something I haven't really fixed for any command yet. same thing with the press command

dallmeyer commented on resources/twitchcommands.py:
```diff
@@ -676,53 +689,54 @@ def gamecontrol():
-        if PREFIX + "circle" == str(args[0]).lower():
-          sendForm("(logior! (cpad-pressed 0) (pad-buttons circle))")
-          message = "" 
+        if PREFIX + "press" == str(args[0]).lower() and len(args) >= 2 and on_check("press") and cd_check("press"):
+            sendForm("(logior! (cpad-pressed 0) (pad-buttons " + str(args[1]) + "))")
+            message = ""
```
similar here, if they give a bad button name hopefully it just does nothing?

dallmeyer replied:
could also be cool to support mapping a single character to the button for shorter commands (with the `press` prefix still)

[c]ircle
[d]own
[l]eft
[r]ight
[s]quare
[t]riangle
[u]p
[x]

dallmeyer reviewed (commented):
we should update the command list on the README too, once these changes are finalized

https://github.com/Zedb0T/Opengoal-Twitch-CrowdControl/blob/main/README.md

MikeGamePro committed:
Checkpoint fix, nojumps + lang, adjustments

- Fixed a misnamed checkpoint (snow-flut-flut)
- Added commands 'nojumps' and 'lang'
- 'press' and 'cam' now have arrays to check for valid arguments, as well as 'lang'
- Removed 'normalcam'
- Changed 'stickycam' to use circular cam instead of fixed for more visibility; also removed zooming for the entirety of the command duration
- Removed the check to see if a command is already active before activating again, to allow more flexibility and to properly reset durations
- Added alternate command names
- env file default value adjustments

dallmeyer commented on resources/twitchcommands.py:
```diff
@@ -192,13 +192,17 @@ def max_val(val, min, max):
 
+lang_list = ["english","french","german","spanish","italian","japanese","uk-english"]
+input_list = ["square","circle","x","triangle","up","down","left","right"]
+cam_list = ["endlessfall","eye","standoff","bike","stick"]
```
Do we need `string` to replace `normalcam`? Or I guess the idea is that the player has to wait out the duration to get back to normal cam?

MikeGamePro replied:
normalcam was used to put it back to normal after invertcam. I removed it because it became kind of useless since it fixes itself after the duration ends. And yeah it goes back to string after the cam command ends, but adding string in the list isn't a bad idea if you want to help the player out and you have the cooldown set to lower than the duration for some reason

Zedb0T commented:
Looks good to me
